src/EnergyPlus/GroundHeatExchangers.cc

```diff
@@ -577,12 +563,17 @@ std::shared_ptr<GLHEVertProps> GetVertProps(EnergyPlusData &state, std::string c
 std::shared_ptr<GLHEVertSingle> GetSingleBH(EnergyPlusData &state, std::string const &objectName)
 {
     // Check if this instance of this model has already been retrieved
-    for (auto &thisBH : state.dataGroundHeatExchanger->singleBoreholesVector) {
-        // Check if the type and name match
-        if (objectName == thisBH->name) {
-            return thisBH;
-        }
-    }
```

Comment:
[src/EnergyPlus/GroundHeatExchangers.cc:582]:(style),[useStlAlgorithm],Consider using std::find_if algorithm instead of a raw loop.